add.php

```diff
@@ -6,12 +6,43 @@
 
 if (isset($_POST['name'])) {
     $name = cleanedInput($_POST['name']);
+    if (strlen($name) > 30) {
+        header('Location: animals.php?error=Invalid name');
+        exit();
+    }
     $status = cleanedInput($_POST['status']);
+    if (strlen($status) > 30) {
+        header('Location: animals.php?error=Invalid status');
+        exit();
+    }
     $weight = cleanedInput($_POST['weight']);
+    if ($weight < 0 || $weight > 10000) {
+        header('Location: animals.php?error=Invalid weight');
+        exit();
+    }
+
     $threat = cleanedInput($_POST['threat']);
+    if (strlen($threat) > 40) {
+        header('Location: animals.php?error=Invalid threat');
+        exit();
+    }
+
     $diet = cleanedInput($_POST['diet']);
+    if (strlen($diet) > 30) {
+        header('Location: animals.php?error=Invalid diet');
+        exit();
+        }
+
     $distribution = cleanedInput($_POST['distribution']);
+    if (strlen($distribution) > 50) {
+        header('Location: animals.php?error=Invalid origin');
+        exit();
+    }
     $image = cleanedInput($_POST['image']);
+    if (strlen($image) > 500) {
+        header('Location: animals.php?error=Invalid image');
+        exit();
+                }
```

Comment:
One final thing! 

Worth adding an extra check to these to make sure the strlen of the inputs is greater than 0 - this accounts for any user who only types a space into one of the fields.

For example:

```
if (strlen($name) > 30 && strlen($name) == 0 {
        header('Location: animals.php?error=Invalid name');
        exit();
    }
```